pkg/ingest/config.go

```diff
@@ -32,6 +32,10 @@ const (
 	// in the worst case scenario, which is expected to be way above the actual one.
 	maxProducerRecordDataBytesLimit = producerBatchMaxBytes - 16384
 	minProducerRecordDataBytesLimit = 1024 * 1024
+
+	// PartitionRingKey is the key under which we store the partitions ring used by the "ingest storage".
+	PartitionRingKey  = "ingester-partitions"
```

Comment:
@mapno Can you give thoughts here?  I think this is not maintaining the distinction between `ingester` and `ingest`, and what "owns" the partitions, that you would know more about.

Reply:
What owns the partitions is whatever component is in the partition ring. This change allows to use a different ring name/key, so multiple partition rings can co-exist and the generator can watch one or the other.

From this PR I understand the idea is to isolate the two parts: the ring and the ring watcher. Tempo's partition ring will continue to be owned by the ingesters. The generator now should be able to watch another ring that's not the one owned by the ingesters. It's confusing.

Reply:
> The generator now should be able to watch another ring that's not the one owned by the ingesters.

Exactly, that's one of the two high-level things this PR is trying to achieve (the other being running a subset of processors only).